soweego/importer/base_dump_extractor.py

```diff
@@ -33,3 +37,23 @@ def get_dump_download_urls(self) -> Iterable[str]:
         :rtype: str
         """
         raise NotImplementedError
+
+    def _commit_entity(self, db_manager, entity, delay: int = 10):
+        success = True
+        session = db_manager.new_session()
+        try:
+            session.add(entity)
+            session.commit()
+        except Exception as ex:
+            LOGGER.error("Failed to commit % s due to %s" % (entity, ex))
+            session.rollback()
+            success = False
+        finally:
+            session.close()
+
+        if not success:
+            LOGGER.info("Sleeping for %s seconds before retrying commit" % delay)
+            time.sleep(delay)
+            delay *= 2
```

Comment:
Why are you doubling the delay here?

Reply:
It's a heuristic. Retrying the failed operation spamming isn't useful.